### ISSUE TITLE
ActionPerformed Fired when setSelection method called in WebSwitch

### ISSUE THREAD

**Issue by iamchathu** [mgarin/weblaf#400]
I need to use two WebSwitchs in a Window where only one can selected at a time. So I have written a toggle method to invert the selection and called in two switchs actionPerformed event in a actionListener.

Problem is setSelection method runs actionPerformed event and StackOverFlow is happening since both of them toggling their states recursively

**Comment by mgarin** [comment]
If you look at any Swing component - they all fire appropriate events even if you set the value through the code, otherwise it is impossible to track changes made in the component settings or data anyhow. Here is an actual example - `JComboBox.setSelectedItem`:
```java
    public void setSelectedItem(Object anObject) {
        Object oldSelection = selectedItemReminder;
        Object objectToSelect = anObject;
        if (oldSelection == null || !oldSelection.equals(anObject)) {

            if (anObject != null && !isEditable()) {
                // For non editable combo boxes, an invalid selection
                // will be rejected.
                boolean found = false;
                for (int i = 0; i < dataModel.getSize(); i++) {
                    E element = dataModel.getElementAt(i);
                    if (anObject.equals(element)) {
                        found = true;
                        objectToSelect = element;
                        break;
                    }
                }
                if (!found) {
                    return;
                }
            }

            // Must toggle the state of this flag since this method
            // call may result in ListDataEvents being fired.
            selectingItem = true;
            dataModel.setSelectedItem(objectToSelect);
            selectingItem = false;

            if (selectedItemReminder != dataModel.getSelectedItem()) {
                // in case a users implementation of ComboBoxModel
                // doesn't fire a ListDataEvent when the selection
                // changes.
                selectedItemChanged();
            }
        }
        fireActionEvent();
    }
```
As you can see it fires both events on the `JComboBox`. You can find similar code pieces in many other basic Swing components.

If you want to do what you mentioned in the topic - you need to ensure that second action event is ignored while first one is being handled and vice versa. I have a lot of similar workarounds for some file chooser events for example.

That said, I will be doing some adjustments to `WebSwitch` component later to enable styling on it and along with those changes I might add a separate kind of events to be fired only when changes are made from the UI - usually that would be an action event. And then you will have another kind of event - change event for example - that will always notify you, doesn't matter where changes came from. That is not something I will do before other important tasks though, so it might take some time.

**Comment by mgarin** [comment]
`WebSwitch` changes didn't make it into v1.2.9 update, but will become available in one of the next smaller updates.

**Comment by mgarin** [comment]
I'll move `WebSwitch` revamp to v1.2.11 for now as it is a fairy small component with just some basic functonality, should fit into the scope.

**Comment by mgarin** [comment]
I've re-checked how some other Swing components solve this problem and seems I was wrong before. I've added a solution similar to `JToggleButton` - `ActionListener` will not fire events anymore when selection is changed from the code, but newly added `ItemListener` will if you need to.

I've pushed the change into git and it will be available with v1.2.11 release soon.
It should also be available in SNAPSHOT repository in a few minutes.

**Comment by mgarin** [comment]
Also since this is solved - less critical `WebSwitch` revamp ( #582 ) will be moved to v1.2.12.
